WordPress/Classes/ViewRelated/Post/PostCell.swift

```diff
@@ -0,0 +1,271 @@
+    @IBOutlet weak var featuredImageStackView: UIStackView!
+    @IBOutlet weak var featuredImage: CachedAnimatedImageView!
+    @IBOutlet weak var featuredImageHeight: NSLayoutConstraint!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var snippetLabel: UILabel!
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var authorLabel: UILabel!
+    @IBOutlet weak var statusLabel: UILabel!
+    @IBOutlet weak var statusView: UIStackView!
+    @IBOutlet weak var progressView: UIProgressView!
+    @IBOutlet weak var editButton: UIButton!
+    @IBOutlet weak var retryButton: UIButton!
+    @IBOutlet weak var viewButton: UIButton!
+    @IBOutlet weak var moreButton: UIButton!
+    @IBOutlet weak var actionBarView: UIStackView!
+    @IBOutlet weak var containerView: UIView!
+    @IBOutlet weak var upperBorder: UIView!
+    @IBOutlet weak var bottomBorder: UIView!
+    @IBOutlet weak var topPadding: NSLayoutConstraint!
+    @IBOutlet weak var contentStackView: UIStackView!
+    @IBOutlet weak var titleAndSnippetView: UIStackView!
+    @IBOutlet weak var topMargin: NSLayoutConstraint!
+
+    private let margin: CGFloat = WPDeviceIdentification.isiPad() ? 20 : 16
+    private let titleTopMargin: CGFloat = WPDeviceIdentification.isiPad() ? 6 : 2
+    private let featuredImageHeightConstant: CGFloat = WPDeviceIdentification.isiPad() ? 226 : 100
+
+    lazy var imageLoader: ImageLoader = {
+        return ImageLoader(imageView: featuredImage, gifStrategy: .mediumGIFs)
+    }()
+
+    private var post: Post!
+    private var viewModel: PostCardStatusViewModel!
+    private weak var interactivePostViewDelegate: InteractivePostViewDelegate?
+    private weak var actionSheetDelegate: PostActionSheetDelegate?
+    var isAuthorHidden: Bool = false {
+        didSet {
+            authorLabel.isHidden = isAuthorHidden
+        }
+    }
+
+    func configure(with post: Post) {
+        if post != self.post {
+            viewModel = PostCardStatusViewModel(post: post)
+        }
+
+        self.post = post
+
+        configureFeaturedImage()
+        configureTitleAndSnippetView()
+        configureTitle()
+        configureSnippet()
+        configureDate()
+        configureAuthor()
+        configureStatusLabel()
+        configureProgressView()
+        configureActionBar()
+    }
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        applyStyles()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if previousTraitCollection?.preferredContentSizeCategory != traitCollection.preferredContentSizeCategory {
+            applyStyles()
+        }
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        imageLoader.prepareForReuse()
+        setNeedsDisplay()
+    }
+
+    override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
```

Comment:
Can you explain the need for a `hitTest`?

Reply:
Done.